vms/evm/sync/customrawdb/accessors_state_sync_test.go

```diff
@@ -0,0 +1,507 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package customrawdb
+
+import (
+	"math/big"
+	"slices"
+	"testing"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/core/rawdb"
+	"github.com/ava-labs/libevm/ethdb"
+	"github.com/ava-labs/libevm/params"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClearAllSyncSegments(t *testing.T) {
+	db := rawdb.NewMemoryDatabase()
+	// add a key that should be cleared
+	require.NoError(t, WriteSyncSegment(db, common.Hash{1}, common.Hash{}))
+
+	// add a key that should not be cleared
+	key := slices.Concat(syncSegmentsPrefix, []byte("foo"))
+	require.NoError(t, db.Put(key, []byte("bar")))
+
+	require.NoError(t, ClearAllSyncSegments(db))
+
+	count := 0
+	it := db.NewIterator(syncSegmentsPrefix, nil)
+	defer it.Release()
+	for it.Next() {
+		count++
+	}
+	require.NoError(t, it.Error())
+	require.Equal(t, 1, count)
+}
+
+func TestWriteReadSyncRoot(t *testing.T) {
+	db := rawdb.NewMemoryDatabase()
+
+	// No root written yet
+	root, err := ReadSyncRoot(db)
+	require.ErrorIs(t, err, ErrEntryNotFound)
+	require.Equal(t, common.Hash{}, root)
+
+	// Write and read back
+	want := common.HexToHash("0x01")
+	require.NoError(t, WriteSyncRoot(db, want))
+	got, err := ReadSyncRoot(db)
+	require.NoError(t, err)
+	require.Equal(t, want, got)
+}
+
+func TestCodeToFetchIteratorAndDelete(t *testing.T) {
+	db := rawdb.NewMemoryDatabase()
+
+	h1 := common.HexToHash("0x11")
+	h2 := common.HexToHash("0x22")
+
+	require.NoError(t, WriteCodeToFetch(db, h1))
+	require.NoError(t, WriteCodeToFetch(db, h2))
+
+	// Insert a malformed key that should be ignored by the iterator (wrong length)
+	bad := append(append([]byte{}, CodeToFetchPrefix...), append(h1.Bytes(), 0x00)...)
+	require.NoError(t, db.Put(bad, []byte("x")))
+
+	seen := map[common.Hash]bool{}
```

Comment:
nit: the `bool` value in this map is actually redundant since the key is sufficient to know if it's in the map or not, so we'd want `map[T]struct{}`, which we have the alias `set.Set` for